core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -543,6 +543,30 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
       }
     }
 
+    if (contains("spark.cores.max")) {
```

Comment:
@srowen  Users may set these configuration via SparkConf object in a programmatic way, can I move the checkings from spark–submit to here, or removed it here directly,which is better?